addons/misra.py

```diff
@@ -3185,7 +3187,36 @@ def misra_17_3(self, cfg):
         for w in cfg.clang_warnings:
             if w['message'].endswith('[-Wimplicit-function-declaration]'):
                 self.reportError(cppcheckdata.Location(w), 17, 3)
-
+        for token in cfg.tokenlist:
+            if token.str not in ["while","if"]:
+                continue
+            if token.next.str != "(":
+                continue
+            tok = token.next
+            while (tok != token.next.link):
+                if tok.isName and not tok.function and tok.next.str == "(":
+                    self.reportError(tok, 17, 3)
+                    break
+                tok = tok.next
+    def misra_config(self,data):
+        for token in data.tokenlist:
+            if token.str not in ["while","if"]:
+                continue
+            if token.next.str != "(":
+                continue
+            tok = token.next
+            while (tok != token.next.link):
+                if tok.str == "(" and tok.isCast:
+                    tok = tok.link
+                    continue
+                if not tok.isName or tok.function or tok.variable or tok.varId or tok.valueType or tok.next.str == "(" \
+                        or tok.str in ["errno","EOF"]:
+                    tok = tok.next
+                    continue
+                errmsg = tok.str + " Variable is unknown"
+                # cppcheckdata.reportError(tok, "style", errmsg, 'misra', "config", [])
+                self.reportError(token, 0, 0,"misra-config")
```

Comment:
I think the "id" should just be "config".. and the "misra-" part will be added by cppcheck etc..